mmv1/third_party/terraform/tests/resource_compute_router_nat_test.go.erb

```diff
@@ -210,6 +210,45 @@ func TestAccComputeRouterNat_withDisabledIndependentEndpointMapping(t *testing.T
 	})
 }
 
+func TestAccComputeRouterNat_withDisabledDynamicPortAllocation(t *testing.T) {
+  t.Parallel()
+
+  testId := randString(t, 10)
+  routerName := fmt.Sprintf("tf-test-router-nat-%s", testId)
+
+  vcrTest(t, resource.TestCase{
+    PreCheck:     func() { testAccPreCheck(t) },
+    Providers:    testAccProviders,
+    CheckDestroy: testAccCheckComputeRouterNatDestroyProducer(t),
+    Steps: []resource.TestStep{
+      {
+        Config: testAccComputeRouterNatWithDisabledDynamicPortAllocation(routerName, true),
+      },
+      {
+        ResourceName:      "google_compute_router_nat.foobar",
+        ImportState:       true,
+        ImportStateVerify: true,
+      },
+      {
+        Config: testAccComputeRouterNatWithDisabledDynamicPortAllocation(routerName, false),
+      },
+      {
+        ResourceName:      "google_compute_router_nat.foobar",
+        ImportState:       true,
+        ImportStateVerify: true,
+      },
+      {
+        Config: testAccComputeRouterNatWithDisabledDynamicPortAllocation(routerName, true),
+      },
+      {
+        ResourceName:      "google_compute_router_nat.foobar",
+        ImportState:       true,
+        ImportStateVerify: true,
+      },
```

Comment:
If the field is not updatable, we probably do need the update steps (the last two config steps) for this test, as setting `enable_dynamic_port_allocation` from `true` to false and then back to `true` would just recreate the resource (delete and create a new resource).
A single step would be enough to test on these two method (`CREATE` and `DELETE`).